src/riak_cs_storage.erl

```diff
@@ -78,21 +78,36 @@ maybe_sum_bucket(User, ?RCS_BUCKET{name=Name} = _Bucket) when is_binary(Name) ->
             {Name, iolist_to_binary(io_lib:format("~p", [E]))}
     end.
 
-%% @doc Sum the number of bytes stored in active files in the named
-%% bucket.  This assumes that the bucket exists; there will be no
-%% difference in output between a non-existent bucket and an empty
-%% one.
+%% @doc Calculate summary for the bucket.
+%%
+%% Basic calculation includes the count and the number of total bytes
+%%  for available files in the named bucket. Detailed one adds some
+%%  more summary information for the bucket.  This assumes that the
+%%  bucket exists; there will be no difference in output between a
+%%  non-existent bucket and an empty one.
 %%
 %% The result is a mochijson structure with two fields: `Objects',
 %% which is the number of objects that were counted in the bucket, and
-%% `Bytes', which is the total size of all of those objects.
--spec sum_bucket(binary()) -> {struct, [{binary(), integer()}]}
-                                   | {error, term()}.
-sum_bucket(BucketName) ->
-    Query = [{map, {modfun, riak_cs_storage, object_size_map},
-              [do_prereduce], false},
-             {reduce, {modfun, riak_cs_storage, object_size_reduce},
-              none, true}],
+%% `Bytes', which is the total size of all of those objects.  More
+%% fields are included for detailed calculation.
+-spec sum_bucket(binary(), boolean()) -> {struct, [{binary(), integer()}]}
+                                             | {error, term()}.
+sum_bucket(BucketName, Detailed) ->
+    Query = case Detailed of
+                false ->
+                    [{map, {modfun, riak_cs_storage, object_size_map},
+                      [do_prereduce], false},
+                     {reduce, {modfun, riak_cs_storage, object_size_reduce},
+                      none, true}];
+                true ->
+                    Now = riak_cs_utils:second_resolution_timestamp(os:timestamp()),
+                    DivPointTs = Now - riak_cs_gc:leeway_seconds(),
+                    DivPoint = {DivPointTs div 1000000, DivPointTs rem 1000000, 0},
+                    [{map, {modfun, riak_cs_storage_mr, bucket_summary_map},
+                      [do_prereduce, {div_point, DivPoint}], false},
```

Comment:
For my personal view, `div_point` is a little bit unclear name. I'd say "leeway_tick", "leeway_period_end" or so.